fleet/metrics.go

```diff
@@ -0,0 +1,212 @@
+package fleet
+
+import (
+	"context"
+	"github.com/go-kit/kit/metrics"
+	"time"
+)
+
+var _ AgentCommsService = (*commsMetricsMiddleware)(nil)
+
+type commsMetricsMiddleware struct {
+	counter metrics.Counter
```

Comment:
these should be named according to the metric itself, because we will have more of these over time. so "request_counter" not just counter, and "request_latency" instead of just latency